From f807d9180fee4e700b042d6baabac7e981570ee4 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Fri, 1 Feb 2019 19:16:35 +0530
Subject: [PATCH 0017/1921] octeontx2-af: Setup resource limits before enabling
 VFs

Initialize resource limits before enabling AF's VFs.
Otherwise after enabling if VFs immediately request for resource
attach then it will fail.

Change-Id: I5f0869bdbce34a111a36861fa0539e48c40f1c96
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu.c | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
index 1ae41370e379..b920eff54d38 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
@@ -2457,19 +2457,18 @@ static int rvu_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 	if (err)
 		goto err_flr;
 
-	/* Enable AF's VFs (if any) */
-	err = rvu_enable_sriov(rvu);
+	err = rvu_policy_init(rvu);
 	if (err)
 		goto err_irq;
 
-	err = rvu_policy_init(rvu);
+	/* Enable AF's VFs (if any) */
+	err = rvu_enable_sriov(rvu);
 	if (err)
-		goto err_sriov;
+		goto err_policy;
 
 	return 0;
-
-err_sriov:
-	rvu_disable_sriov(rvu);
+err_policy:
+	rvu_policy_destroy(rvu);
 err_irq:
 	rvu_unregister_interrupts(rvu);
 err_flr:
-- 
2.31.1

